saleor/graphql/checkout/tests/mutations/test_checkout_complete_price_override.py

```diff
@@ -0,0 +1,742 @@
+def test_checkout_complete_with_price_override_and_voucher_specific_product(
+    price_override,
+    reward,
+    user_api_client,
+    checkout_with_item,
+    address,
+    shipping_method,
+    voucher_specific_product_type,
+):
+    # given
+    voucher = voucher_specific_product_type
+    voucher_code = voucher.codes.first().code
+    voucher_channel_listings = voucher.channel_listings.first()
+    voucher_channel_listings.discount_value = reward
+    voucher_channel_listings.save(update_fields=["discount_value"])
+
+    checkout = checkout_with_item
+    checkout.shipping_address = address
+    checkout.shipping_method = shipping_method
+    checkout.billing_address = address
+    checkout.tax_exemption = True
+    checkout.voucher_code = voucher_code
+    checkout.save()
+
+    line = checkout.lines.first()
+    line.price_override = price_override
+    line.save(update_fields=["price_override"])
+    quantity = line.quantity
+
+    channel = checkout.channel
+    channel.allow_unpaid_orders = True
+    channel.save(update_fields=["allow_unpaid_orders"])
+
+    manager = get_plugins_manager(allow_replica=False)
+    lines, _ = fetch_checkout_lines(checkout)
+    checkout_info = fetch_checkout_info(checkout, lines, manager)
+    add_voucher_code_to_checkout(manager, checkout_info, lines, voucher_code)
+
+    shipping_price = shipping_method.channel_listings.get().price_amount
+    unit_discount = Decimal(price_override * reward / 100)
+    base_unit_price = price_override - unit_discount
+    subtotal = base_unit_price * quantity
+    total = subtotal + shipping_price
+    undiscounted_subtotal = Decimal(price_override * quantity)
+    undiscounted_total = undiscounted_subtotal + shipping_price
+
+    redirect_url = "https://www.example.com"
+    variables = {"id": to_global_id_or_none(checkout), "redirectUrl": redirect_url}
+
+    # when
+    response = user_api_client.post_graphql(MUTATION_CHECKOUT_COMPLETE, variables)
+
+    # then
+    content = get_graphql_content(response)
+    data = content["data"]["checkoutComplete"]
+    assert not data["errors"]
+
+    order = Order.objects.get()
+    assert order.status == OrderStatus.UNCONFIRMED
+    assert order.origin == OrderOrigin.CHECKOUT
+    assert order.subtotal_net_amount == subtotal
+    assert order.shipping_price_net_amount == shipping_price
+    assert order.total_net_amount == total
+    assert order.undiscounted_total_net_amount == undiscounted_total
+
+    order_line = order.lines.get()
+    assert order_line.is_price_overridden
+    assert order_line.undiscounted_base_unit_price_amount == price_override
+    assert order_line.base_unit_price_amount == base_unit_price
+    assert order_line.total_price_net_amount == subtotal
+    assert order_line.undiscounted_total_price_net_amount == undiscounted_subtotal
+    assert order_line.unit_discount_amount == unit_discount
+
+    # TODO: fix OrderDiscount object values for specific product voucher with overriden
```

Comment:
Is it TODO for later? If yes maybe worth to have this inside a task?

Reply:
Here is internal issue: https://linear.app/saleor/issue/SHOPX-1283